### PR TITLE
feat: rewrite entire code base to vite open-sauced

### LINKED ISSUE #80: Feature: clean up tooling and remove astro (breaking)
### Type of feature

Build

### Current behavior

We are unable to use [astro](https://astro.build) in several instances:
- code listing and formatting (due to missing eslint parser and language tools)
- cloud IDE providers (due to astro vs vite config mismatch)
- language completion and assistance in local IDEs (language tools need manual install and are not bundled with astro)
- production builds due to unresolved bugs (see #76)
- the technology is in perpetual alpha/beta since 2017
- [partial hydration](https://docs.astro.build/en/core-concepts/component-hydration/) the only feature we are missing, is not fully working and requires additional code

### Desired solution

1. strip `.astro` files into `.jsx`
1. apply vite bundler configuration from open-sauced so we can contribute without local install
1. replace head component with hydrated react-helmet or similar
1. switch to pre-rendered static content globally
1. apply service worker

### Additional context

doing this now will fix a lot of headaches for existing and future contributors

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by 0-vortex**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [x] 🔥 Performance Improvements
- [ ] ✅ Test
- [x] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR removes astro and rewrites to simplified [open-sauced/open-sauced](https://github.com/open-sauced/open-sauced) vite template.

For this to properly record a major version increase we need to ensure the `BREAKING CHANGE: removed astro` text appears in the squashed commit - or you can let @0-vortex do the actual merge upon proper review.

**Current Issues**:
- ~~font-awesome icons styling on header~~
- eslint vite cache gets stuck sometimes
- ~~used <any> in some vague places~~
- **no onegraph stuff, at all**

**TODO**:
- vite pwa configuration
- incomplete postcss configuration/packages
- ~~enforce type checking~~
- ~~split components better~~

## Related Tickets & Documents

BREAKING CHANGE: removed astro
closes #80

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [x] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [x] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

- improve on the baseline
- implement alpha?/beta?/next branch
- improve on the vite template

### CODE REVIEW

**Comment by bdougie** [comment]
note: Deploy preview is broken because of the VITE_ prefix change.

File: src/components/PostGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import Avatar from './Avatar';
+import { updateVotesByRepo } from '../lib/supabase';
+import { getRepoLink } from '../lib/github';
+import useSupabaseAuth from '../hooks/useSupabaseAuth';
+import { User } from "@supabase/supabase-js";
+import { FaArrowAltCircleUp } from "react-icons/fa";
+
+export declare interface PostGridProps {
+  data: any;
```
Comment (bdougie) [inline src/components/PostGrid.tsx:10]: I am not a strong TypeScript user, but the use of "any" here seems to be raising some flags.

File: src/lib/contributorHelper.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Lifted from:
+// https://stackoverflow.com/questions/34954347/using-reduce-to-build-a-filter-function-in-javascript
+function reduce(array: string[], combine: (array: string[], value: any) => string[], start: string[]) {
```
Comment (bdougie) [inline src/lib/contributorHelper.ts:3]: value should be an integer or number.

File: src/lib/supabase.ts [hunk 3]
```diff
@@ -1,32 +1,10 @@
-import { createClient } from '@supabase/supabase-js';
+import {createClient, User} from '@supabase/supabase-js';
 
-// probably should move these to an env.
-export const supabase = createClient(import.meta.env.PUBLIC_SUPABASE_URL,
-  import.meta.env.PUBLIC_SUPABASE_API_KEY);
+export const supabase = createClient(
+  import.meta.env.VITE_SUPABASE_URL,
+  import.meta.env.VITE_SUPABASE_API_KEY);
```
Comment (bdougie) [inline src/lib/supabase.ts:5]: Should make this a 2.0.0 change with this. Is it possible to implement a setup script to copy a gist for the env setup? Just thinking of thoughts on how to mitigate this breaking change. 

I suppose we can share a note in the discord too.

**Review by bdougie — COMMENTED** [review]
The majority looks good. 

I like the switch to TypeScript in this and appreciate you taking that on. Curious if we should be looking into Vitest in a future PR and thoughts on the post-deploy plan for the breaking change.

**Comment by 0-vortex** [comment]
> note: Deploy preview is broken because of the VITE_ prefix change. @0-vortex you mentioned this in the Discord, but wanted to leave a note for context sharing.
> 
> Also what is handling the local redirect when logging in?
> 
> I also created a new GitHub OAuth secret and will delete the old one when this is merged.

We need both apps, one is set up for the production database and one for us to commit to local supabase toml configuration. Both of them should exist, only one of them known and the other kept secret - if attackers try to exploit the OAuth key they should be able to simply disrupt local development. For local redirect there is already a Vite base URL variable we could be using, but at the moment I don't think it's handled :D

**Comment by 0-vortex** [comment]
> The majority looks good.
> 
> I like the switch to TypeScript in this and appreciate you taking that on. Curious if we should be looking into Vitest in a future PR and thoughts on the post-deploy plan for the breaking change.

Tested Vitest for the Webpack rewrite in open-sauced, am a big fan of @antfu's stuff but this one is far from production-ready. Breaking change is just different env variables and would like to commit with working local development - just add the new variables on the pipeline and maybe cover/commit #139 first. Want to force a major version tag because rewrite, not actually breaking changes per say :D

File: src/lib/supabase.ts (hunk 3 above)
Comment (0-vortex) [reply src/lib/supabase.ts:5]: Gist is my least favorite method as leaking that we wouldn't be able to tell what happened. New variables will be locked in for local development, I just copy-pasted the wrong `.gitignore` :D. Just add the new variables and all should work, delete the old ones once this is merged

File: src/lib/contributorHelper.ts (hunk 2 above)
Comment (0-vortex) [reply src/lib/contributorHelper.ts:3]: adjusting this shortly 🍕

**Review by bdougie — APPROVED** [review]

File: src/lib/supabase.ts (hunk 3 above)
Comment (0-vortex) [reply src/lib/supabase.ts:5]: reverted the BC on this, we have the same limitations as open-sauced and previous hot instance :D

**Comment by 0-vortex** [comment]
Unfortunately, deploy previews have to be manually added until  https://github.com/supabase/supabase/issues/4808 is merged.

**Pull request merged**